scripts/src/javatests/org/oppia/android/scripts/maven/GenerateLicenseTextsTest.kt

```diff
@@ -0,0 +1,259 @@
+package org.oppia.android.scripts.maven
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.oppia.android.scripts.proto.License
+import org.oppia.android.scripts.proto.MavenDependency
+import org.oppia.android.scripts.proto.MavenDependencyList
+import org.oppia.android.scripts.proto.PrimaryLinkType
+import org.oppia.android.testing.assertThrows
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.PrintStream
+
+/** Tests for [GenerateLicenseTexts]. */
+class GenerateLicenseTextsTest {
+  private val TOO_LESS_ARGS_FAILURE = "Too less arguments passed"
+  private val MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE_FAILURE =
+    "maven_dependencies.textproto is not up-to-date"
+  private val SCRIPT_PASSED_INDICATOR = "Script execution completed."
+  private val VALID_LINK = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+  private val INVALID_LINK = "https://fabric.io/terms"
```

Comment:
Please move the HTTP download portion of the script to a utility that's abstracted by an interface (maybe LicenseFetcher?) so that we can mock it out in the test. We should never be relying on network connectivity during a test since that both introduces flakiness and can prevent people from developing the codebase while offline.

Reply:
Done.